src/components/DragAndDrop/DragAndDrop.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button } from 'antd';
+import React, { useState } from 'react';
+import { BsPlusLg } from 'react-icons/bs';
+
+interface IDragAndDropProps {
+  image: string | ArrayBuffer | null;
+  setImage: (a: string | ArrayBuffer | null) => any;
+  maxFileSize: number;
+}
+
+export const DragAndDrop: React.FC<IDragAndDropProps> = ({
+  image,
+  setImage,
+  maxFileSize = 10000000000000,
+}) => {
+  const [drag, setDrag] = useState<boolean>(false);
+  const [fileError, setFileError] = useState<string>('');
+
+  const toBase64 = (file: File): Promise<string | ArrayBuffer | null> =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = () => resolve(reader.result);
+      reader.onerror = (error) => reject(error);
+    });
+
+  const dragStartHandler = (e: any) => {
+    e.preventDefault();
+    setDrag(true);
+  };
+
+  const dragLeaveHandler = (e: any) => {
+    e.preventDefault();
+    setDrag(false);
+  };
+
+  const dropHandler = async (e: any) => {
+    e.preventDefault();
+    const files = [...e.dataTransfer.files];
+    if (
+      (files[0].type === 'image/jpeg' || files[0].type === 'image/png') &&
+      files[0].size < maxFileSize
+    ) {
+      const fileBase64 = await toBase64(files[0]);
+      setImage(fileBase64);
+      setFileError('');
+    } else {
+      setFileError(
+        `Please send png or jpeg file and that file's size should be ${maxFileSize}`,
+      );
+    }
+    setDrag(false);
+  };
+
+  return (
+    <div className='flex flex-col'>
+      <div className='flex items-center dark:bg-black'>
+        <div className='flex flex-col items-center h-44 w-28 justify-between'>
+          <h4 className='text-center dark:text-white'>Drop your avatar here</h4>
```

Comment:
This should be dynamic . f.e. `label` prop

Reply:
https://github.com/igorizviekov/mt-booking_app/pull/40/commits/ac85c3c8b18d02bdf3af1b8d93764919140e815b